### PR TITLE
bcc: 0.26.0 -> 0.28.0

### PR DESCRIPTION
**Pull request by martinetd**
###### Description of changes

- upgrade bcc from 0.26 to 0.28. The python packaging changed a bit but it apparently just requires pulling in setuptools and fixing up PYTHONPATH like some other services did. I've renamed python to python3 just because I don't see the point of keeping it as a variable now python = python2 is dead.
- while here upgrade the LLVM toolchain used by bcc/bpftrace to the latest llvm 16; happy to split that out if appropriate I've tested the build with both the previous llvm14 and this.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/bcc/default.nix [hunk 1]
```diff
@@ -31,8 +31,11 @@ python.pkgs.buildPythonApplication rec {
     ./fix-deadlock-detector-import.patch
   ];
 
-  propagatedBuildInputs = [ python.pkgs.netaddr ];
-  nativeBuildInputs = [ makeWrapper cmake flex bison llvmPackages.llvm.dev ];
+  propagatedBuildInputs = [ python3.pkgs.netaddr ];
+  nativeBuildInputs = [
+    makeWrapper cmake flex bison llvmPackages.llvm.dev zip
+    python3.pkgs.setuptools
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/bcc/default.nix:37]:
```suggestion
    bison
    cmake
    flex
    llvmPackages.llvm.dev
    makeWrapper
    python3.pkgs.setuptools
    zip
```

File: pkgs/os-specific/linux/bcc/default.nix [hunk 2]
```diff
@@ -59,6 +62,10 @@ python.pkgs.buildPythonApplication rec {
       --replace '$'{exec_prefix}/@CMAKE_INSTALL_LIBDIR@ @CMAKE_INSTALL_FULL_LIBDIR@
   '';
 
+  preInstall = ''
+    # required for setuptool during install
+    export PYTHONPATH=$out/${python3.sitePackages}:$PYTHONPATH
+  '';
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/bcc/default.nix:68]: This could probably be fixed by constructing a python env

**Comment by martinetd** [comment]
(trivial rebase to rerun tests, failed on some github internal error...)

**Comment by martinetd** [comment]
(removed unused libelf argument as done by @Mic92 in his PR)

File: pkgs/os-specific/linux/bcc/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/os-specific/linux/bcc/default.nix:86]: > /home/x/nixpkgs/outputs/out/lib/python3.10/site-packages/

What is that path? Are you not using the sandbox?

Comment (martinetd) [reply pkgs/os-specific/linux/bcc/default.nix:86]: I was building with `nix -L build -f . bcc` from nixpkgs' directory; `/home/x/nixpkgs` was the nixpkgs directory (I removed the user bit), on a nixos unstable linux machine.

I'm almost certain I didn't disable the sandbox (I actually have `nix.settings.sandbox = true;` from before it was default but I think it's not required anymore now...?) , but I guess it's the path that was derived from whatever cmake install target the build script derived.
But yeah I'm surprised it contained my home path and not /build or whatever builder sandbox path should be using...

I retried building just now, printing `$out` and it appears correct, and so is the python path printed when I built just now which has the same warnings with paths I'd expect:
```
bcc> $out is /nix/store/ijf86wzk41fdkly3slfvkr71dv6lbfk6-bcc-0.28.0
[...]
bcc> Checking .pth file support in /nix/store/ijf86wzk41fdkly3slfvkr71dv6lbfk6-bcc-0.28.0/lib/python3.10/site-packages/
bcc> /nix/store/1r6n7v2wam7gkr18gxccpg7p5ywgw551-python3-3.10.12/bin/python3 -E -c pass
bcc> TEST FAILED: /nix/store/ijf86wzk41fdkly3slfvkr71dv6lbfk6-bcc-0.28.0/lib/python3.10/site-packages/ does NOT support .pth files
bcc> bad install directory or PYTHONPATH
bcc> You are attempting to install a package to a directory that is not
bcc> on PYTHONPATH and which Python does not read ".pth" files from.  The
bcc> installation directory you specified (via --install-dir, --prefix, or
bcc> the distutils default setting) was:
bcc>     /nix/store/ijf86wzk41fdkly3slfvkr71dv6lbfk6-bcc-0.28.0/lib/python3.10/site-packages/
bcc> and your PYTHONPATH environment variable currently contains:
bcc>     '/nix/store/1r6n7v2wam7gkr18gxccpg7p5ywgw551-python3-3.10.12/lib/python3.10/site-packages:/nix/store/cr689gv3clfqhbl05pfqd9g9khzdidcb-python3.10-setuptools-67.4.0/lib/python3.10/site-packages:/nix/store/mss4z8ivvpl5px7jvwhb8gfxs7m1c0ym-python3.10-netaddr-0.8.0/lib/python3.10/site-packages'
bcc> Here are some of your options for correcting the problem:
bcc> * You can choose a different installation directory, i.e., one that is
bcc>   on PYTHONPATH or supports .pth files
bcc> * You can add the installation directory to the PYTHONPATH environment
bcc>   variable.  (It must then also be on PYTHONPATH whenever you run
bcc>   Python and want to use the package(s) you are installing.)
bcc> * You can set up the installation directory to support ".pth" files by
bcc>   using one of the approaches described here:
bcc>   https://setuptools.pypa.io/en/latest/deprecated/easy_install.html#custom-installation-locations
```

So, err, not quite sure what happened, and any logs that might have remained on filesystem have long been gc'd so /shrug
(I also tried various commits from reflog to check and not noticing a difference, but that won't be downgrading my 'nix' version to the os I was running at the time)


Either way the error remains the same: without setting the variable, the package builds but the python install step failed and the resulting egg file is unusable from python import directives

Comment (Mic92) [reply pkgs/os-specific/linux/bcc/default.nix:86]: I would be fine with this solution. python env might be actually more code and currently I don't see the advantage of it.

**Review by Mic92 — APPROVED** [review]

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 240520` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>bcc (linuxKernel.packages.hardkernel_4_14.bcc ,linuxKernel.packages.linux_4_14.bcc ,linuxKernel.packages.linux_4_14_hardened.bcc ,linuxKernel.packages.linux_4_19.bcc ,linuxKernel.packages.linux_4_19_hardened.bcc ,linuxKernel.packages.linux_5_10.bcc ,linuxKernel.packages.linux_5_10_hardened.bcc ,linuxKernel.packages.linux_5_15.bcc ,linuxKernel.packages.linux_5_15_hardened.bcc ,linuxKernel.packages.linux_5_4.bcc ,linuxKernel.packages.linux_5_4_hardened.bcc ,linuxKernel.packages.linux_6_1.bcc ,linuxKernel.packages.linux_6_1_hardened.bcc ,linuxKernel.packages.linux_6_3.bcc ,linuxKernel.packages.linux_6_3_hardened.bcc ,linuxKernel.packages.linux_6_4.bcc ,linuxKernel.packages.linux_hardened.bcc ,linuxKernel.packages.linux_latest_libre.bcc ,linuxKernel.packages.linux_libre.bcc ,linuxKernel.packages.linux_lqx.bcc ,linuxKernel.packages.linux_testing_bcachefs.bcc ,linuxKernel.packages.linux_xanmod.bcc ,linuxKernel.packages.linux_xanmod_latest.bcc ,linuxKernel.packages.linux_xanmod_stable.bcc ,linuxKernel.packages.linux_zen.bcc ,linuxPackages-libre.bcc ,linuxPackages.bcc ,linuxPackages_4_14_hardened.bcc ,linuxPackages_4_19_hardened.bcc ,linuxPackages_5_10_hardened.bcc ,linuxPackages_5_15_hardened.bcc ,linuxPackages_5_4_hardened.bcc ,linuxPackages_6_1_hardened.bcc ,linuxPackages_6_3_hardened.bcc ,linuxPackages_hardened.bcc ,linuxPackages_hardkernel_latest.bcc ,linuxPackages_latest-libre.bcc ,linuxPackages_latest.bcc ,linuxPackages_lqx.bcc ,linuxPackages_testing_bcachefs.bcc ,linuxPackages_xanmod.bcc ,linuxPackages_xanmod_latest.bcc ,linuxPackages_xanmod_stable.bcc ,linuxPackages_zen.bcc)</li>
    <li>bcc.man (linuxKernel.packages.hardkernel_4_14.bcc.man ,linuxKernel.packages.linux_4_14.bcc.man ,linuxKernel.packages.linux_4_14_hardened.bcc.man ,linuxKernel.packages.linux_4_19.bcc.man ,linuxKernel.packages.linux_4_19_hardened.bcc.man ,linuxKernel.packages.linux_5_10.bcc.man ,linuxKernel.packages.linux_5_10_hardened.bcc.man ,linuxKernel.packages.linux_5_15.bcc.man ,linuxKernel.packages.linux_5_15_hardened.bcc.man ,linuxKernel.packages.linux_5_4.bcc.man ,linuxKernel.packages.linux_5_4_hardened.bcc.man ,linuxKernel.packages.linux_6_1.bcc.man ,linuxKernel.packages.linux_6_1_hardened.bcc.man ,linuxKernel.packages.linux_6_3.bcc.man ,linuxKernel.packages.linux_6_3_hardened.bcc.man ,linuxKernel.packages.linux_6_4.bcc.man ,linuxKernel.packages.linux_hardened.bcc.man ,linuxKernel.packages.linux_latest_libre.bcc.man ,linuxKernel.packages.linux_libre.bcc.man ,linuxKernel.packages.linux_lqx.bcc.man ,linuxKernel.packages.linux_testing_bcachefs.bcc.man ,linuxKernel.packages.linux_xanmod.bcc.man ,linuxKernel.packages.linux_xanmod_latest.bcc.man ,linuxKernel.packages.linux_xanmod_stable.bcc.man ,linuxKernel.packages.linux_zen.bcc.man ,linuxPackages-libre.bcc.man ,linuxPackages.bcc.man ,linuxPackages_4_14_hardened.bcc.man ,linuxPackages_4_19_hardened.bcc.man ,linuxPackages_5_10_hardened.bcc.man ,linuxPackages_5_15_hardened.bcc.man ,linuxPackages_5_4_hardened.bcc.man ,linuxPackages_6_1_hardened.bcc.man ,linuxPackages_6_3_hardened.bcc.man ,linuxPackages_hardened.bcc.man ,linuxPackages_hardkernel_latest.bcc.man ,linuxPackages_latest-libre.bcc.man ,linuxPackages_latest.bcc.man ,linuxPackages_lqx.bcc.man ,linuxPackages_testing_bcachefs.bcc.man ,linuxPackages_xanmod.bcc.man ,linuxPackages_xanmod_latest.bcc.man ,linuxPackages_xanmod_stable.bcc.man ,linuxPackages_zen.bcc.man)</li>
    <li>bpftrace (linuxKernel.packages.hardkernel_4_14.bpftrace ,linuxKernel.packages.linux_4_14.bpftrace ,linuxKernel.packages.linux_4_14_hardened.bpftrace ,linuxKernel.packages.linux_4_19.bpftrace ,linuxKernel.packages.linux_4_19_hardened.bpftrace ,linuxKernel.packages.linux_5_10.bpftrace ,linuxKernel.packages.linux_5_10_hardened.bpftrace ,linuxKernel.packages.linux_5_15.bpftrace ,linuxKernel.packages.linux_5_15_hardened.bpftrace ,linuxKernel.packages.linux_5_4.bpftrace ,linuxKernel.packages.linux_5_4_hardened.bpftrace ,linuxKernel.packages.linux_6_1.bpftrace ,linuxKernel.packages.linux_6_1_hardened.bpftrace ,linuxKernel.packages.linux_6_3.bpftrace ,linuxKernel.packages.linux_6_3_hardened.bpftrace ,linuxKernel.packages.linux_6_4.bpftrace ,linuxKernel.packages.linux_hardened.bpftrace ,linuxKernel.packages.linux_latest_libre.bpftrace ,linuxKernel.packages.linux_libre.bpftrace ,linuxKernel.packages.linux_lqx.bpftrace ,linuxKernel.packages.linux_testing_bcachefs.bpftrace ,linuxKernel.packages.linux_xanmod.bpftrace ,linuxKernel.packages.linux_xanmod_latest.bpftrace ,linuxKernel.packages.linux_xanmod_stable.bpftrace ,linuxKernel.packages.linux_zen.bpftrace ,linuxPackages-libre.bpftrace ,linuxPackages.bpftrace ,linuxPackages_4_14_hardened.bpftrace ,linuxPackages_4_19_hardened.bpftrace ,linuxPackages_5_10_hardened.bpftrace ,linuxPackages_5_15_hardened.bpftrace ,linuxPackages_5_4_hardened.bpftrace ,linuxPackages_6_1_hardened.bpftrace ,linuxPackages_6_3_hardened.bpftrace ,linuxPackages_hardened.bpftrace ,linuxPackages_hardkernel_latest.bpftrace ,linuxPackages_latest-libre.bpftrace ,linuxPackages_latest.bpftrace ,linuxPackages_lqx.bpftrace ,linuxPackages_testing_bcachefs.bpftrace ,linuxPackages_xanmod.bpftrace ,linuxPackages_xanmod_latest.bpftrace ,linuxPackages_xanmod_stable.bpftrace ,linuxPackages_zen.bpftrace)</li>
    <li>bpftrace.man (linuxKernel.packages.hardkernel_4_14.bpftrace.man ,linuxKernel.packages.linux_4_14.bpftrace.man ,linuxKernel.packages.linux_4_14_hardened.bpftrace.man ,linuxKernel.packages.linux_4_19.bpftrace.man ,linuxKernel.packages.linux_4_19_hardened.bpftrace.man ,linuxKernel.packages.linux_5_10.bpftrace.man ,linuxKernel.packages.linux_5_10_hardened.bpftrace.man ,linuxKernel.packages.linux_5_15.bpftrace.man ,linuxKernel.packages.linux_5_15_hardened.bpftrace.man ,linuxKernel.packages.linux_5_4.bpftrace.man ,linuxKernel.packages.linux_5_4_hardened.bpftrace.man ,linuxKernel.packages.linux_6_1.bpftrace.man ,linuxKernel.packages.linux_6_1_hardened.bpftrace.man ,linuxKernel.packages.linux_6_3.bpftrace.man ,linuxKernel.packages.linux_6_3_hardened.bpftrace.man ,linuxKernel.packages.linux_6_4.bpftrace.man ,linuxKernel.packages.linux_hardened.bpftrace.man ,linuxKernel.packages.linux_latest_libre.bpftrace.man ,linuxKernel.packages.linux_libre.bpftrace.man ,linuxKernel.packages.linux_lqx.bpftrace.man ,linuxKernel.packages.linux_testing_bcachefs.bpftrace.man ,linuxKernel.packages.linux_xanmod.bpftrace.man ,linuxKernel.packages.linux_xanmod_latest.bpftrace.man ,linuxKernel.packages.linux_xanmod_stable.bpftrace.man ,linuxKernel.packages.linux_zen.bpftrace.man ,linuxPackages-libre.bpftrace.man ,linuxPackages.bpftrace.man ,linuxPackages_4_14_hardened.bpftrace.man ,linuxPackages_4_19_hardened.bpftrace.man ,linuxPackages_5_10_hardened.bpftrace.man ,linuxPackages_5_15_hardened.bpftrace.man ,linuxPackages_5_4_hardened.bpftrace.man ,linuxPackages_6_1_hardened.bpftrace.man ,linuxPackages_6_3_hardened.bpftrace.man ,linuxPackages_hardened.bpftrace.man ,linuxPackages_hardkernel_latest.bpftrace.man ,linuxPackages_latest-libre.bpftrace.man ,linuxPackages_latest.bpftrace.man ,linuxPackages_lqx.bpftrace.man ,linuxPackages_testing_bcachefs.bpftrace.man ,linuxPackages_xanmod.bpftrace.man ,linuxPackages_xanmod_latest.bpftrace.man ,linuxPackages_xanmod_stable.bpftrace.man ,linuxPackages_zen.bpftrace.man)</li>
    <li>oci-seccomp-bpf-hook (linuxPackages.oci-seccomp-bpf-hook)</li>
    <li>oci-seccomp-bpf-hook.man (linuxPackages.oci-seccomp-bpf-hook.man)</li>
    <li>linuxPackages.system76-scheduler (linuxPackages-libre.system76-scheduler ,linuxPackages_4_14_hardened.system76-scheduler ,linuxPackages_4_19_hardened.system76-scheduler ,linuxPackages_5_10_hardened.system76-scheduler ,linuxPackages_5_15_hardened.system76-scheduler ,linuxPackages_5_4_hardened.system76-scheduler ,linuxPackages_6_1_hardened.system76-scheduler ,linuxPackages_6_3_hardened.system76-scheduler ,linuxPackages_hardened.system76-scheduler ,linuxPackages_hardkernel_latest.system76-scheduler ,linuxPackages_latest-libre.system76-scheduler ,linuxPackages_latest.system76-scheduler ,linuxPackages_lqx.system76-scheduler ,linuxPackages_testing_bcachefs.system76-scheduler ,linuxPackages_xanmod.system76-scheduler ,linuxPackages_xanmod_latest.system76-scheduler ,linuxPackages_xanmod_stable.system76-scheduler ,linuxPackages_zen.system76-scheduler)</li>
    <li>picosnitch</li>
    <li>picosnitch.dist</li>
    <li>sockdump</li>
  </ul>
</details>

**Pull request merged**